client/network/src/lib.rs

```diff
@@ -454,3 +457,106 @@ async fn wait_to_announce<Block: BlockT>(
 		);
 	}
 }
+
+#[derive(Clone)]
+pub struct WaitForParachainTargetBlock<Block> {
+	phantom: PhantomData<Block>,
+}
+
+impl<Block: BlockT> WaitForParachainTargetBlock<Block> {
+	/// Get warp sync target block
+	pub async fn warp_sync_get(
+		para_id: ParaId,
+		relay_chain_interface: Arc<dyn RelayChainInterface>,
+		spawner: Arc<dyn SpawnNamed + Send + Sync>,
+	) -> Result<oneshot::Receiver<Block::Header>, BoxedError>
+	where
+		Block: BlockT + 'static,
+	{
+		let (sender, receiver) = oneshot::channel::<Block::Header>();
+
+		spawner.spawn(
+			"cumulus-parachain-wait-for-target-block",
+			None,
+			async move {
+				tracing::info!(
+					target: LOG_TARGET,
+					"waiting for target block in a background task...",
+				);
+				Self::wait_for_target_block(sender, para_id, relay_chain_interface).await;
+				tracing::info!(target: LOG_TARGET, "target block reached",);
+			}
+			.boxed(),
+		);
+
+		return Ok(receiver)
+	}
+
+	async fn wait_for_target_block(
+		sender: oneshot::Sender<Block::Header>,
+		para_id: ParaId,
+		relay_chain_interface: Arc<dyn RelayChainInterface>,
+	) {
+		let import_stream = relay_chain_interface.import_notification_stream();
+		match import_stream.await {
+			Ok(mut import_stream_notification) => match import_stream_notification.next().await {
+				Some(_header) => loop {
+					let is_syncing = relay_chain_interface
+						.is_major_syncing()
+						.await
+						.map_err(|e| {
+							tracing::error!(
+								target: LOG_TARGET,
+								"Unable to determine sync status. {}",
+								e
+							)
+						})
+						.unwrap_or(false);
+
+					if !is_syncing {
```

Comment:
Again unneeded indentation.

Reply:
I kept the variable to keep the code readable.